File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> m1, m2;
        for(char c: s) {
            m1[c] += 1;
        }

        for(char c: t) {
            m2[c] += 1;
        }

        for(auto i: m1) {
            if(m2[i.first] != i.second) {
                return false;
            } else {
                m1[i.first] = -1;
                m2[i.first] = -1;
            }
        }

        for(auto i: m1) {
            if(i.second != -1) return false;
        }

        for(auto i: m2) {
            if(i.second != -1) return false;
        }

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) < len(t):
            s, t = t, s
        s_map = defaultdict(int)
        t_map = defaultdict(int)
        for c in s:
            s_map[c]+=1
        for c in t:
            t_map[c]+=1
        
        for c in s_map:
            if c not in t_map:
                return False
            if s_map[c] != t_map[c]:
                return False
        return True